docusaurus/docs/create-a-plugin/extend-a-plugin/instrument-a-backend-plugin.md

```diff
@@ -0,0 +1,421 @@
+
+### Metric types
+
+See [Prometheus metric types](https://prometheus.io/docs/concepts/metric_types/) for a list and description of the different metric types you can use and when to use them.
+
+There are many possible types of metrics that can be tracked. One popular method for defining metrics is the [RED method](https://grafana.com/blog/2018/08/02/the-red-method-how-to-instrument-your-services/).
+
+### Implement metrics in your plugin
+
+[Grafana plugin SDK for Go](../../introduction/grafana-plugin-sdk-for-go.md) uses the [Prometheus instrumentation library for Go applications](https://github.com/prometheus/client_golang). Any custom metric registered with the [default registry](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#pkg-variables) will be picked up by the SDK and exposed through the [Collect metrics capability](../../introduction/backend.md#collect-metrics).
+
+For convenience, It's recommended to use the [promauto package](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promauto) when creating custom metrics since it will automatically register the metric in the [default registry](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#pkg-variables).
+
+**Example:**
+
+```go
+package plugin
+
+import (
+    "context"
+
+    "github.com/prometheus/client_golang/prometheus"
+    "github.com/prometheus/client_golang/prometheus/promauto"
+
+    "github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+var queriesTotal = promauto.NewCounterVec(
+    prometheus.CounterOpts{
+        Namespace: "grafana_plugin",
+        Name:      "query",
+        Help:      "Total number of queries.",
+    },
+    []string{"query_type"},
+)
+
+func (ds *Datasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+    for _, q := range req.Queries {
+        queriesTotal.WithLabelValues(q.QueryType).Inc()
+    }
+}
+```
+
+### Best practices/recommendations
+
+- Consider using the namespace _grafana_plugin_ as that would prefix any defined metric names with plugin*. This will make it clear for operators that any metric named \*grafana_plugin*\* originates from a Grafana plugin.
+- Use snake case style when naming metrics, e.g. _http_request_duration_seconds_ instead of _httpRequestDurationSeconds_.
+- Use snake case style when naming metric labels, e.g. _status_code_ instead of _statusCode_.
+- If the metric type is a counter, name it with a _\_total suffix_, e.g. _http_requests_total_.
+- If the metric type is a histogram and you're measuring duration, name it with a _`_<unit>`_ suffix, e.g. _http_request_duration_seconds_.
+- If the metric type is a gauge, name it to denote it's a value that can increase and decrease , e.g. _http_request_in_flight_.
```

Comment:
I'm a bit confused by the markdown here

![immagine](https://github.com/grafana/plugin-tools/assets/16373015/c62cd30e-3035-4644-83c1-51e9f5a76ee4)


Maybe you wanted something like:

```suggestion
- Consider using the namespace `grafana_plugin` as that would prefix any defined metric names with `plugin`. This will make it clear for operators that any metric named `grafana_plugin` originates from a Grafana plugin.
- Use snake case style when naming metrics, e.g. `http_request_duration_seconds` instead of `httpRequestDurationSeconds`.
- Use snake case style when naming metric labels, e.g. `status_code` instead of `statusCode`.
- If the metric type is a counter, name it with a `_total` suffix, e.g. `http_requests_total`.
- If the metric type is a histogram and you're measuring duration, name it with a `_<unit>`  suffix, e.g. `http_request_duration_seconds`.
- If the metric type is a gauge, name it to denote it's a value that can increase and decrease , e.g. `http_request_in_flight`.
```

:point_down: 

![immagine](https://github.com/grafana/plugin-tools/assets/16373015/06551223-ecd6-4985-866a-01cc9bf2fe94)